filters/include/pcl/filters/lambda_filter_indices.h

```diff
@@ -0,0 +1,166 @@
+/*
+ * Software License Agreement (BSD License)
+ *
+ *  Point Cloud Library (PCL) - www.pointclouds.org
+ *  Copyright (c) 2020-, Open Perception
+ *
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of the copyright holder(s) nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *
+ * $Id$
+ *
+ */
+
+#pragma once
+
+#include <pcl/filters/filter_indices.h>
+#include <pcl/pcl_macros.h>
+#include <pcl/type_traits.h> // for is_invocable
+
+#ifndef __cpp_lib_is_invocable
+#include <boost/hof/is_invocable.hpp>
+#endif
+
+#include <type_traits>
+
+#include <utility>
+
+namespace pcl {
+namespace detail {
+template <typename PointT, typename Function>
+using PointFilterLambda =
+    std::enable_if_t<pcl::is_invocable_r_v<bool,
+                                           Function,
+                                           const pcl::remove_cvref_t<PointT>&,
+                                           pcl::index_t>,
+                     bool>;
```

Comment:
1. This should not be a type. It should be a free function which returns true or false in case the supplied functor object cannot be called with required function signature. 

2. Assuming you don't define this, and the user supplies a bad functor, the compiler will still complain and not compile. What is the point of defining this instead of letting the compiler simply do its job?

Reply:
1. Converted to constexpr variable
2. Introspection for others, better error message, constrain to non-modifying lambda (forces the lambda to not take input by reference and thus prevent potential modifications)